comparison/comparison.go

```diff
@@ -0,0 +1,509 @@
+// Copyright (c) 2019-present Mattermost, Inc. All Rights Reserved.
```

Comment:
It crossed my mind to put it there but since it's such a high level construct, making use of pretty much all top level packages, it felt strange to have it buried inside a specific one.